addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
@stefanpenner - This is required until I make the wait helper in integration tests handle all promises (as you mentioned the wait helper in acceptance tests already does this). The gain is fairly significant (that we can use ember-ajax from component integration tests), and I don't see a significant downside. 

I'm happy to coordinate be general RSVP awareness in the ember-test-helpers wait helper so that we can remove this wrapping, but in the meantime I think we should land it...